pkg/cmd/kafka/create/api_validators.go

```diff
@@ -0,0 +1,134 @@
+	"github.com/redhat-developer/app-services-cli/pkg/shared/accountmgmtutil"
+	"github.com/redhat-developer/app-services-cli/pkg/shared/connection"
+	"github.com/redhat-developer/app-services-cli/pkg/shared/factory"
+	"github.com/redhat-developer/app-services-cli/pkg/shared/remote"
+	kafkamgmtclient "github.com/redhat-developer/app-services-sdk-go/kafkamgmt/apiv1/client"
+	"k8s.io/utils/strings/slices"
+)
+
+type ValidatorInput struct {
+	provider string
+	region   string
+	size     string
+
+	userAMSInstanceType *accountmgmtutil.QuotaSpec
+
+	f         *factory.Factory
+	constants *remote.DynamicServiceConstants
+	conn      connection.Connection
+}
+
+func (input *ValidatorInput) ValidateProviderAndRegion() error {
+	f := input.f
+	f.Logger.Debug("Validating provider and region")
+	cloudProviders, _, err := input.conn.API().
+		KafkaMgmt().
+		GetCloudProviders(f.Context).
+		Execute()
+
+	if err != nil {
+		return err
+	}
+
+	var selectedProvider kafkamgmtclient.CloudProvider
+
+	providerNames := make([]string, 0)
+	for _, item := range cloudProviders.Items {
+		if !item.GetEnabled() {
+			continue
+		}
+		if item.GetId() == input.provider {
+			selectedProvider = item
+		}
+		providerNames = append(providerNames, item.GetId())
+	}
+	f.Logger.Debug("Validating cloud provider", input.provider, ". Enabled providers: ", providerNames)
+
+	if !selectedProvider.Enabled {
+		providers := strings.Join(providerNames, ",")
+		providerEntry := localize.NewEntry("Provider", input.provider)
+		validProvidersEntry := localize.NewEntry("Providers", providers)
+		return f.Localizer.MustLocalizeError("kafka.create.provider.error.invalidProvider", providerEntry, validProvidersEntry)
+	}
+
+	return validateProviderRegion(input, selectedProvider)
+}
+
+func validateProviderRegion(input *ValidatorInput, selectedProvider kafkamgmtclient.CloudProvider) error {
+	f := input.f
+	cloudRegion, _, err := input.conn.API().
+		KafkaMgmt().
+		GetCloudProviderRegions(f.Context, selectedProvider.GetId()).
+		Execute()
+
+	if err != nil {
+		return err
+	}
+
+	var selectedRegion kafkamgmtclient.CloudRegion
+	regionNames := make([]string, 0)
+	for _, item := range cloudRegion.Items {
+		if !item.GetEnabled() {
+			continue
+		}
+		regionNames = append(regionNames, item.GetId())
+		if item.GetId() == input.region {
+			selectedRegion = item
+		}
+	}
+
+	if len(regionNames) != 0 {
+		f.Logger.Debug("Validating region", input.region, ". Enabled regions: ", regionNames)
+		regionsString := strings.Join(regionNames, ", ")
+		if !selectedRegion.Enabled {
+			regionEntry := localize.NewEntry("Region", input.region)
+			validRegionsEntry := localize.NewEntry("Regions", regionsString)
+			providerEntry := localize.NewEntry("Provider", input.provider)
+			return f.Localizer.MustLocalizeError("kafka.create.region.error.invalidRegion", regionEntry, providerEntry, validRegionsEntry)
+		}
+
+		return nil
+
+		// Current API doesn't allow us to validate regions properly.
```

Comment:
Not sure if we should handle region out of the capacity problem or leave it for the API result. 

It is possible to implement this but logic will be quite tricky and subject to breaking changes

Reply:
Backend will also check if the instance type is supported in the region and if there is capacity.

Reply:
Yes. I think while UI for this is not great. I would ignore that use case for both clients and 100% rely on the backend error code. That will save us some problems with breaking this model once we get more instance types/sizes etc.